Prospress/ruleset.xml

```diff
@@ -2,105 +2,24 @@
-		</properties>
-	</rule>
-
-	<rule ref="Generic.Files.EndFileNewline"/>
+	<!-- Import the main WordPress rules -->
+	<rule ref="WordPress-Core"/>
 
-	<!-- https://make.wordpress.org/core/handbook/coding-standards/php/#naming-conventions -->
-	<rule ref="Generic.Files.LowercasedFilename"/>
-
-	<!-- https://make.wordpress.org/core/handbook/coding-standards/php/#space-usage -->
-	<rule ref="Generic.Formatting.SpaceAfterCast"/>
-
-	<!-- https://make.wordpress.org/core/handbook/coding-standards/php/#brace-style -->
-	<rule ref="Generic.Functions.OpeningFunctionBraceKernighanRitchie"/>
-
-	<rule ref="PEAR.Functions.FunctionCallSignature">
-		<properties>
-			<property name="requiredSpacesAfterOpen" value="1" />
-			<property name="requiredSpacesBeforeClose" value="1" />
-		</properties>
-	</rule>
-	<rule ref="PEAR.Functions.FunctionCallSignature.ContentAfterOpenBracket">
-		<severity>0</severity>
-	</rule>
-	<rule ref="PEAR.Functions.FunctionCallSignature.CloseBracketLine">
-		<severity>0</severity>
-	</rule>
+	<!-- Selectively import other WordPress rules -->
+	<rule ref="WordPress.PHP.DiscouragedFunctions"/>
+	<rule ref="WordPress.Security.NonceVerification"/>
+	<rule ref="WordPress.WP.EnqueuedResources"/>
+	<rule ref="WordPress.Variables.GlobalVariables"/>
 
-	<rule ref="WordPress.Arrays.ArrayDeclaration">
-		<exclude name="WordPress.Arrays.ArrayDeclaration.SingleLineNotAllowed" />
+	<!-- Don't check the number of new lines after control structures -->
+	<rule ref="Squiz.ControlStructures.ControlSignature">
+		<exclude name="Squiz.ControlStructures.ControlSignature.NewlineAfterOpenBrace"/>
 	</rule>
-	<rule ref="WordPress.Arrays.ArrayKeySpacingRestrictions"/>
-	<rule ref="WordPress.Classes.ValidClassName"/>
-	<rule ref="WordPress.Files.FileName"/>
-	<rule ref="WordPress.Functions.FunctionDeclarationArgumentSpacing"/>
-	<rule ref="WordPress.NamingConventions.ValidFunctionName"/>
-	<rule ref="WordPress.WhiteSpace.ControlStructureSpacing"/>
-	<rule ref="WordPress.WhiteSpace.OperatorSpacing"/>
-	<rule ref="WordPress.WhiteSpace.CastStructureSpacing"/>
-	<rule ref="WordPress.PHP.YodaConditions"/>
 
-	<rule ref="WordPress.XSS.EscapeOutput">
-		<!-- ignore wcs plain text email templates - strip_tags is run over the output amongst other things -->
-		<exclude-pattern>*/templates/emails/plain/*</exclude-pattern>
+	<!-- Don't require class files to start with "class-" -->
+	<rule ref="WordPress.Files.FileName">
 		<properties>
-			<!-- e.g. body_class, the_content, the_excerpt -->
-			<property name="customAutoEscapedFunctions" type="array" value="0=>woocommerce_wp_select,1=>wcs_help_tip"/>
-			<!-- e.g. esc_attr, esc_html, esc_url-->
-			<property name="customEscapingFunctions" type="array" value="0=>wcs_json_encode,1=>htmlspecialchars,2=>wp_kses_allow_underscores"/>
-			<!-- e.g. _deprecated_argument, printf, _e-->
-			<property name="customPrintingFunctions" type="array" value=""/>
+			<property name="strict_class_file_names" value="false"/>
```

Comment:
❔ As far as I have observed in WCS code, I see class file names starting with "class-". Any reason why this has not been enforced strictly?

Reply:
Not all of our libraries use the `class-` prefix in file naming.